src/modm/architecture/interface/can_message.hpp

```diff
@@ -92,17 +92,53 @@ struct Message
 	uint8_t length;
 
 public:
-	bool
-	operator == (const modm::can::Message& rhs) const;
+	inline bool
+	operator == (const modm::can::Message& rhs) const
+	{
+		if ((this->identifier     == rhs.identifier) and
+			(this->length         == rhs.length)     and
+			(this->flags.rtr      == rhs.flags.rtr)  and
+			(this->flags.extended == rhs.flags.extended))
+		{
+			bool eq = true;
+			for (uint8_t ii = 0; ii < this->length; ++ii)
+			{
+				if (this->data[ii] != rhs.data[ii]) {
+					eq = false;
+				}
+			}
+			return eq;
```

Comment:
These loop could be replaced by one call to `std::equal(data, data + length, rhs.data)`.